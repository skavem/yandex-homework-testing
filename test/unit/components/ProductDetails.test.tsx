import React from "react";
import renderer from "react-test-renderer";
import { CartApi, ExampleApi } from "../../../src/client/api";
import { ProductDetails } from "../../../src/client/components/ProductDetails";
import { initStore } from "../../../src/client/store";
import __Providers from "../__Providers";
import { basename, getPath } from "../helpers";

describe("ProductDetails", () => {
  it("сохраняет верстку", () => {
    const api = new ExampleApi(basename);
    const cart = new CartApi();
    const store = initStore(api, cart);

    const app = (
      <__Providers
        api={api}
        cart={cart}
        store={store}
        basename={getPath("/catalog/0", true)}
      >
        <ProductDetails
          product={{
            color: "red",
            id: 1,
            name: "test",
            price: 10,
            description: "lorem ipsum",
            material: "copium",
          }}
        />
      </__Providers>
    );

    const tree = renderer.create(app).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
