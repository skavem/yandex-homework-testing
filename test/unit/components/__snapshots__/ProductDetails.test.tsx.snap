// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetails рендерится корректно 1`] = `
<div
  className="ProductDetails row"
>
  <div
    className="col-12 col-sm-5 col-lg-4"
  >
    <img
      className="Image"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkMAYAADkANVKH3ScAAAAASUVORK5CYII="
    />
  </div>
  <div
    className="col-12 col-sm-7 col-lg-6"
  >
    <h1
      className="ProductDetails-Name"
    >
      test
    </h1>
    <p
      className="ProductDetails-Description"
    >
      lorem ipsum
    </p>
    <p
      className="ProductDetails-Price fs-3"
    >
      $
      10
    </p>
    <p>
      <button
        className="ProductDetails-AddToCart btn btn-primary btn-lg"
        onClick={[Function]}
      >
        Add to Cart
      </button>
    </p>
    <dl>
      <dt>
        Color
      </dt>
      <dd
        className="ProductDetails-Color text-capitalize"
      >
        red
      </dd>
      <dt>
        Material
      </dt>
      <dd
        className="ProductDetails-Material text-capitalize"
      >
        copium
      </dd>
    </dl>
  </div>
</div>
`;

exports[`ProductDetails сохраняет верстку 1`] = `
<div
  className="ProductDetails row"
>
  <div
    className="col-12 col-sm-5 col-lg-4"
  >
    <img
      className="Image"
      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkMAYAADkANVKH3ScAAAAASUVORK5CYII="
    />
  </div>
  <div
    className="col-12 col-sm-7 col-lg-6"
  >
    <h1
      className="ProductDetails-Name"
    >
      test
    </h1>
    <p
      className="ProductDetails-Description"
    >
      lorem ipsum
    </p>
    <p
      className="ProductDetails-Price fs-3"
    >
      $
      10
    </p>
    <p>
      <button
        className="ProductDetails-AddToCart btn btn-primary btn-lg"
        onClick={[Function]}
      >
        Add to Cart
      </button>
    </p>
    <dl>
      <dt>
        Color
      </dt>
      <dd
        className="ProductDetails-Color text-capitalize"
      >
        red
      </dd>
      <dt>
        Material
      </dt>
      <dd
        className="ProductDetails-Material text-capitalize"
      >
        copium
      </dd>
    </dl>
  </div>
</div>
`;
